portal-impl/src/com/liferay/portal/service/impl/RoleLocalServiceImpl.java

```diff
@@ -682,6 +682,15 @@ else if (group.isLayout() || group.isLayoutSetPrototype() ||
 	}
 
 	@Override
+	public int getRegularRoleUsersGroupsCount(Role role) {
```

Comment:
If you need to check the role type (or any other field) just grab it from the db. Most probably it will be feched from the cache it, so the cost will be very small.

Also, see my other comment on merging the count methods.